server/game/timeLimit.js

```diff
@@ -23,25 +23,25 @@ class TimeLimit {
         if(!this.timeLimitStarted) {
             this.timeLimitStarted = true;
             this.timeLimitStartedAt = new Date();
-            this.game.addMessage('Time limit of {0} minutes starts now!', this.timeLimitInMinutes);
+
+            this.timer = setInterval(() => {
+                this.checkForTimeLimitReached();
+            }, 1000);
         }
     }
 
     checkForTimeLimitReached() {
         if(this.game.useGameTimeLimit && !this.isTimeLimitReached) {
             let differenceBetweenStartOfTimerAndNow = moment.duration(moment().diff(this.timeLimitStartedAt));
             if(differenceBetweenStartOfTimerAndNow.asSeconds() / 60 >= this.timeLimitInMinutes) {
-                this.game.addMessage('Time limit of {0} minutes reached, the game will end after the current round has finished!', this.timeLimitInMinutes);
+                this.game.addAlert('warning', 'Time up.  The game will end after the current round has finished');
                 this.isTimeLimitReached = true;
-                this.game.on('onRoundEnded', () => this.handleEndOfRoundWhenTimeLimitIsReached());
+                this.timeLimitStarted = false;
+                this.game.timeExpired();

```

Comment:
What's the trade off here between defining `timeExpired` here vs doing the `emit` here? Both require the `game` object. Is there any game specific logic we anticipate in the future that should live on `Game` instead of `TimeLimit`?

Reply:
I felt that it was better abstraction to not have the timer reaching inside of the game state stuff or raising events on his behalf. What do you think?

Reply:
Agree on not reaching in for state. Not raising the event makes `TimeLimit` feel like it's a leaky abstraction, as `Game` has to know how to signal that time has expired. 

Part of the problem is we've conflated game state (e.g. list of players and their state) and game messaging bus (e.g. something happened in the game and other components might want to know) into a single giant object. But that's a larger problem than this PR.